### PR TITLE
FIX: New corrections workflow: items get stuck during corrections workflow [SDCP-551]

### PR DESCRIPTION
**Pull request by GyanP**

### CODE REVIEW

**Review by petrjasek — APPROVED** [review]
would be good to handle those error cases we have so those items are not stuck there

File: apps/archive/archive_correction.py [hunk 1]
```diff
@@ -95,17 +95,23 @@ def on_update(self, updates, original):
...
             # modify item in published.
-            published_service.system_update(
+            _published_item = published_service.system_update(
                 published_article.get(config.ID_FIELD), published_item_updates, published_article
             )
+            if _published_item is None or (
+                published_item_updates.get("operation") == CONTENT_STATE.BEING_CORRECTED
+                and _published_item.get("state") != CONTENT_STATE.BEING_CORRECTED
+            ):
+                raise SuperdeskApiError.badRequestError(message=_("Being corrected item is not generated"))
```
Comment (petrjasek) [inline apps/archive/archive_correction.py:105]: not sure this is bad request error, should be more like assert error

Comment (petrjasek) [reply apps/archive/archive_correction.py:105]: eg:
```py
assert _published_item is not None and (...)
```

Comment (GyanP) [reply apps/archive/archive_correction.py:105]: I think we can use `preconditionFailedError` instead of `badRequestError`  or assert is more convenient?

Comment (petrjasek) [reply apps/archive/archive_correction.py:105]: precondition is used for etag validation, do you think it could be an issue here?

Comment (GyanP) [reply apps/archive/archive_correction.py:105]: No, I think we will not get etag issue. I will add changes with assert.

**Review by petrjasek — APPROVED** [review]

**Pull request merged**